### ISSUE TITLE
[HOLD for payment 2025-01-02] [$250] Android - Workspace - White space below next button & keypad not closed on tapping outside

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53910]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** V9. 0.74-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**Issue reported by:** Applause Internal Team

## Action Performed:
Pre- condition: Have control workspace with 2 members invited, enabled workflow and add approval 
1. Launch app in both mweb and android
2. Go to workspace settings - workflow
3. Tap add approval workflow
4. Note the white space between next button and keypad in mweb and android
5. Tap outside on top
6. Note keypad behavior on mweb and android

## Expected Result:
In add approval workflow page, the white space between next button and keypad must be consistent, similarly on tapping outside keypad closing behavior too must be consistent.

## Actual Result:
In add approval workflow page, the white space between next button and keypad in less in mweb and more in android. After tapping outside, keypad closed in mweb and remain opened in android.

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/a6d362f8-93cd-4bed-ad41-af391a684a20

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021867598579287232853</li>
        <li>Upwork Job ID: 1867598579287232853</li>
        <li>Last Price Increase: 2024-12-13</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @Christinadobrzyn</details>

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
There are extra spaces between the button and keyboard on the workflow page.

### What is the root cause of that problem?
On the Expenses From page, we disable the padding bottom from ScreenWrapper.
https://github.com/Expensify/App/blob/8749259618f3f9c31eb817281aea1f482ca26202/src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsExpensesFromPage.tsx#L205-L207

The padding bottom is added from the SelectionList using `useStyledSafeAreaInsets`.
https://github.com/Expensify/App/blob/8749259618f3f9c31eb817281aea1f482ca26202/src/components/SelectionList/BaseSelectionList.tsx#L770-L774

Also, the button is wrapped with FixedFooter which has its own padding bottom already.
https://github.com/Expensify/App/blob/8749259618f3f9c31eb817281aea1f482ca26202/src/components/SelectionList/BaseSelectionList.tsx#L839-L841

This looks fine when the keyboard isn't visible because the padding fills the device navigation bar space, but when the keyboard shows, it looks like too much space. It doesn't happen on another page that uses SelectionList too because `useStyledSafeAreaInsets` returns 0 padding if it's already applied by the ScreenWrapper.

Why it's not a problem if the padding bottom is added from the ScreenWrapper? It's because the padding bottom is applied to the View that wraps the KeyboardAvoidingView.
https://github.com/Expensify/App/blob/97d59214da66d8f40e808884c491e6c03528de9b/src/components/ScreenWrapper.tsx#L278-L288

But the padding in SelectionList is added as the children of the KeyboardAvoidingView.

### What changes do you think we should make in order to solve the problem?

Enable the padding bottom from the ScreenWrapper for the ExpensesFrom page.
https://github.com/Expensify/App/blob/8749259618f3f9c31eb817281aea1f482ca26202/src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsExpensesFromPage.tsx#L205-L207

(we can check for other pages that uses BaseSelectionList too)

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

N/A

**Comment by Christinadobrzyn** [comment]
Hum this is a really minor change. I'm not sure if this is something that is very important to update at this time. Asking the team.

**Comment by Christinadobrzyn** [comment]
Just a heads up that I'm going to be ooo Dec 12 - 13th. Back on Monday 16th. I'm not going to assign this to a BZ teammate but if anything is urgent, please reach out to the team for a volunteer.

**Comment by bernhardoj** [comment]
It's not something critical, but seeing those extra spaces is very unpleasant. (this happens on some pages too)

**Comment by Christinadobrzyn** [comment]
Okay, I think if you think this should be resolved, we'll move it forward. Adding external for a review of https://github.com/Expensify/App/issues/53910#issuecomment-2536165711.

**Comment by c3024** [comment]
Is this a problem for all pages with `ScreenWrapper` with `includeSafeAreaPaddingBottom` `false` and with `BaseSelectionList ` with search bar enabled as child?

**Comment by bernhardoj** [comment]
Yes, except if the list doesn't have a footer. For example, it happens on the Approver page too (the Approver search input only shows if there is >8 item)

**Comment by bernhardoj** [comment]
PR is ready

cc: @c3024

**Comment by Christinadobrzyn** [comment]
Payment summary here - https://github.com/Expensify/App/issues/53910#issuecomment-2567460622

@c3024 do we need a regression test for this?

**Comment by Christinadobrzyn** [comment]
DMd @c3024 to see if we need a regression test

**Comment by c3024** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [x] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: https://github.com/Expensify/App/pull/37391/files#r1904879930

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: This is a specific edge case and it does not need a discussion.

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


Have a "Control" workspace with 
1. 2 members invited.
2. "Workflows" enabled.
3. "Add approvals" enabled in "Workflows".


### Test:


1. Go to Workspace Settings > Workflows.
2. Press "Add approval workflow".
3. On mobile, make sure the keyboard is shown, if not, press on the input.
4. Verify there are no extra spaces below the button

Do we agree 👍 or 👎


</details>

**Comment by bernhardoj** [comment]
Requested in ND.

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj

**Comment by JmillsExpensify** [comment]
$250 approved for @c3024
